configs/config.go

```diff
@@ -0,0 +1,186 @@
+package config
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/spf13/viper"
+)
+
+type LogConfig struct {
+	Level  string `mapstructure:"level"`
+	Pretty bool   `mapstructure:"pretty"`
+}
+
+type PollerConfig struct {
+	Enabled       bool `mapstructure:"enabled"`
+	Interval      int  `mapstructure:"interval"`
+	BlocksPerPoll int  `mapstructure:"blocks_per_poll"`
+	FromBlock     int  `mapstructure:"from_block"`
+	UntilBlock    int  `mapstructure:"until_block"`
+}
+
+type CommitterConfig struct {
+	Enabled   bool `mapstructure:"enabled"`
+	Interval  int  `mapstructure:"interval"`
+	BatchSize int  `mapstructure:"batch_size"`
+}
+
+type FailureRecovererConfig struct {
+	Enabled   bool `mapstructure:"enabled"`
+	Interval  int  `mapstructure:"interval"`
+	BatchSize int  `mapstructure:"batch_size"`
+}
+
+type StorageConfig struct {
+	Staging      StorageConnectionConfig `mapstructure:"staging"`
+	Main         StorageConnectionConfig `mapstructure:"main"`
+	Orchestrator StorageConnectionConfig `mapstructure:"orchestrator"`
+}
+type StorageType string
+
+const (
+	StorageTypeMain         StorageType = "main"
+	StorageTypeStaging      StorageType = "staging"
+	StorageTypeOrchestrator StorageType = "orchestrator"
+)
+
+type StorageDriver string
+
+const (
+	StorageDriverMemory     StorageDriver = "memory"
+	StorageDriverClickhouse StorageDriver = "clickhouse"
+)
+
+type StorageConnectionConfig struct {
+	Driver     StorageDriver    `mapstructure:"driver"`
+	Clickhouse ClickhouseConfig `mapstructure:"clickhouse"`
+	Memory     MemoryConfig     `mapstructure:"memory"`
+}
+
+type ClickhouseConfig struct {
+	Host     string `mapstructure:"host"`
+	Port     int    `mapstructure:"port"`
+	Username string `mapstructure:"username"`
+	Password string `mapstructure:"password"`
+	Database string `mapstructure:"database"`
+}
+
+type MemoryConfig struct {
+	MaxItems int `mapstructure:"max_items"`
+}
+
+type RPCConfig struct {
+	URL string `mapstructure:"url"`
+}
+
+type Config struct {
+	RPC              RPCConfig              `mapstructure:"rpc"`
+	Log              LogConfig              `mapstructure:"log"`
+	Poller           PollerConfig           `mapstructure:"poller"`
+	Committer        CommitterConfig        `mapstructure:"committer"`
+	FailureRecoverer FailureRecovererConfig `mapstructure:"failure_recoverer"`
+	Storage          StorageConfig          `mapstructure:"storage"`
+}
+
+func LoadConfig(cfgFile string) error {
+	if cfgFile != "" {
+		viper.SetConfigFile(cfgFile)
+		if err := viper.ReadInConfig(); err != nil {
+			return fmt.Errorf("error reading config file, %s", err)
+		}
+	} else {
+		viper.SetConfigName("config")
+		viper.AddConfigPath("./configs")
+
+		if err := viper.ReadInConfig(); err != nil {
+			return fmt.Errorf("error reading config file, %s", err)
+		}
+
+		viper.SetConfigName("secrets")
+		err := viper.MergeInConfig()
```

Comment:
How does this work?how does it know where the secrets file is?

Reply:
the config path is already set before by `viper.AddConfigPath("./configs")`. So it looks for a file with name `secrets` to load the secrets.